rpcserver.go

```diff
@@ -212,40 +236,52 @@ func (s *server) SendCollation(
 		collation.Period,
 		len(collation.Blobs),
 	)
-	// Tag the span with collations info
-	span.SetTag("Shard", collation.ShardID)
-	span.SetTag("Period of collation", collation.Period)
-	span.SetTag("Blobs of collation", collation.Blobs)
+	// Tag collation info if nothing goes wrong
+	logger.SetTag(spanctx, "Shard", collation.ShardID)
+	logger.SetTag(spanctx, "Period of collation", collation.Period)
+	logger.SetTag(spanctx, "Blobs of collation", collation.Blobs)
 	return makePlainResponse(true, replyMsg), nil
 }
 
 func (s *server) StopServer(
 	ctx context.Context,
 	req *pbrpc.RPCStopServerRequest) (*pbrpc.RPCPlainResponse, error) {
 	// Add span for StopServer
-	span := opentracing.StartSpan("RPCServer.StopServer", opentracing.ChildOf(s.parentSpan.Context()))
+	spanctx, err := logger.StartFromParentState(ctx, "RPCServer.StopServer", s.serializedSpanCtx)
+	if err != nil {
+		logger.FinishWithErr(spanctx, fmt.Errorf("Failed to deserialize parent span context, err: %v", err))
+	}
+	defer logger.Finish(spanctx)
 
 	log.Printf("rpcserver:StopServer: receive=%v", req)
-	time.Sleep(time.Millisecond * 500)
-	span.Finish()
 	log.Printf("Closing RPC server by rpc call...")
-	s.rpcServer.Stop()
+	go func() {
+		time.Sleep(time.Second * 1)
+		s.rpcServer.Stop()
+	}()
```

Comment:
What do you think if we use a channel instead of a `time.Sleep`?
```
chanFuncReturned := make(chan struct{})
defer func() {
	chanFuncReturned <- struct{}{}
	
}()
go func() {
	<-chanFuncReturned
	s.rpcServer.Stop()
}()
```

Reply:
As you said, with my approach, `s.rpcServer.Stop()` still seemed to execute before the content is returned. In this case, we can just use `time.Sleep` until we find another better approach